commit e737e7baf547347daeac08b59e80cbad2b1e7e9b
Author: Kelly McLaughlin <kelly.mclaughlin@joyent.com>
Date:   2019-08-22T08:35:42-06:00 (7 weeks ago)
    
    MANTA-4287 Update node-moray to use node-fast 2.7.0
    Reviewed by: Kody A Kantor <kody@kkantor.com>
    Reviewed by: Rui <rui@joyent.com>

diff --git a/lib/client.js b/lib/client.js
index c2c08f0..7077330 100644
--- a/lib/client.js
+++ b/lib/client.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2018, Joyent, Inc.
+ * Copyright 2019 Joyent, Inc.
  */
 
 /*
@@ -88,6 +88,7 @@ function MorayClient(options) {
     assert.optionalBool(options.requireIndexes, 'options.requireIndexes');
     assert.optionalBool(options.requireOnlineReindexing,
         'options.requireOnlineReindexing');
+    assert.optionalNumber(options.crc_mode, 'options.crc_mode');
 
     coptions = parseMorayParameters(options);
     cueballOptions = coptions.cueballOptions;
@@ -100,6 +101,7 @@ function MorayClient(options) {
     this.requireIndexes = options.requireIndexes ? true : false;
     this.requireOnlineReindexing =
         options.requireOnlineReindexing ? true : false;
+    this.crc_mode = options.crc_mode;
 
     /* Helper objects. */
     this.log = options.log.child({
@@ -340,7 +342,8 @@ MorayClient.prototype.createFastConnection =
         'log': this.log.child({
             'component': 'FastClient',
             'backendName': backend.name
-        })
+        }),
+        'crc_mode': this.crc_mode
     }));
 };
 
diff --git a/lib/fast_connection.js b/lib/fast_connection.js
index 8b57811..7d2ebbb 100644
--- a/lib/fast_connection.js
+++ b/lib/fast_connection.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2018, Joyent, Inc.
+ * Copyright 2019 Joyent, Inc.
  */
 
 /*
@@ -52,9 +52,12 @@ function FastConnection(args) {
     assert.number(args.nRecentRequests, 'args.nRecentRequests');
     assert.number(args.tcpKeepAliveInitialDelay,
         'args.tcpKeepAliveInitialDelay');
+    assert.optionalNumber(args.crc_mode, 'args.crc_mode');
 
     events.EventEmitter.call(this);
 
+    this.crc_mode = args.crc_mode || fast.FAST_CHECKSUM_V1;
+
     /* Arguments */
     this.fc_connect_address = args.address;
     this.fc_connect_port = args.port;
@@ -67,7 +70,8 @@ function FastConnection(args) {
         'collector': args.collector, /* Optional: artedi metrics collector */
         'log': args.log,
         'nRecentRequests': args.nRecentRequests,
-        'transport': this.fc_sock
+        'transport': this.fc_sock,
+        'crc_mode': this.crc_mode
     });
 
     /*
diff --git a/package.json b/package.json
index 72f21c1..04a9591 100644
--- a/package.json
+++ b/package.json
@@ -17,7 +17,7 @@
         "cmdutil": "^1.1.0",
         "cueball": "^2.3.0",
         "extsprintf": "^1.3.0",
-        "fast": "2.6.0",
+        "fast": "2.8.1",
         "libuuid": "0.2.1",
         "jsprim": "^1.3.0",
         "posix-getopt": "^1.0.0",
